libutils/sequence.c

```diff
@@ -418,16 +418,17 @@ void *const *SeqGetData(const Seq *seq)
     return seq->data;
 }
 
-void SeqRemoveNulls(Seq *seq)
+Seq *SeqFilter(Seq *const seq, bool (*filter)(void *))
 {
     assert(seq != NULL);
```

Comment:
I'd probably write this to allow passing in NULL and just use the default condition instead of a callback for the simple case, but since you went down the route of `FilterNullCallback()`, I think you should add an assert here instead;

```suggestion
    assert(seq != NULL);
    assert(filter != NULL);
```